src/main/java/io/opentracing/contrib/global/concurrent/TracedCallable.java

```diff
@@ -0,0 +1,122 @@
+
+    private final Callable<T> delegate;
+    private String operationName = "";
+    private SpanContext parentContext = null;
+
+    protected TracedCallable(Callable<T> delegate) {
+        if (delegate == null) throw new NullPointerException("Callable delegate is <null>.");
+        this.delegate = delegate;
+    }
+
+    public static <T> TracedCallable<T> of(Callable<T> delegate) {
+        return new TracedCallable<T>(delegate);
+    }
+
+    public TracedCallable<T> withOperationName(String operationName) {
+        this.operationName = operationName;
+        return this;
+    }
+
+    public TracedCallable<T> asChildOf(SpanContext parent) {
+        this.parentContext = parent;
+        return this;
+    }
+
+    protected Span startNewSpan() {
+        SpanBuilder spanBuilder = GlobalTracer.get().buildSpan(operationName);
+        if (parentContext != null) spanBuilder = spanBuilder.asChildOf(parentContext);
+        return spanBuilder.start();
+    }
+
+    /**
+     * This method {@linkplain #startNewSpan() starts a new span} and calls the delegate, returning the result.
+     * Finally, the started span is {@linkplain Span#close() closed} again.
+     *
+     * @return The result from the delegate call.
+     * @throws Exception in case the delegate call threw an exception
+     *                   or there were exceptions starting or closing a new span.
+     */
+    public T call() throws Exception {
+        // The javadoc doesn't mention the details of the errorhandling
+        // because they are in line with normal try-with-resources logic:
+        // - try to close resources created (started span) whether the call succeeded or failed.
+        // - in case closing fails: a) throw the call exception if there was one (but addSuppressed the close failure)
+        //                       or b) if there was no call exception throw the close failure instead.
+
+        Span newSpan = null;            // so it can be closed
+        Exception callException = null; // so it can be thrown from finally if necessary
+        try {
+
+            newSpan = startNewSpan();
+            return delegate.call();
+
+        } catch (Exception callEx) {
+            callException = callEx;
+            throw callException;
+        } finally {
+            if (newSpan != null) try {
+                newSpan.close();
+            } catch (Exception closeFailure) {
+                //noinspection ThrowFromFinallyBlock
+                throw callException == null ? closeFailure
+                        : ThrowableSupport.addSuppressed(callException, closeFailure); // Don't shadow callException!
+            }
+        }
+    }
+
+    /**
+     * Utility to call Throwable.addSuppressed() method form Java 6 code if it is running in a Java 7 JVM.
+     * If the JVM is 1.6, the 'toBeSuppressed' exception will be logged instead.
+     */
+    private static final class ThrowableSupport {
```

Comment:
ps this is the last comment I'll make on this..

if suppression is a primary feature of this library, it should be tested like one. Guava is far more reviewed than this PR and more used than any library any of us here has created. Reusing some of this not only saves you work, but also it reduces the amount of stuff reviewers like me have to look at. Ex. I don't need to scrutinize tested code in guava.. it is one of the most reused libraries there is.

By reuse I mean you can literally copy/paste the inner types from Closer and cite them, if this feature is critical to you. Then, you can reuse the tests of them. Another alternative is to use something like shade to make closer bundled as an internal type. Then you wouldn't need to do much work except reference x.y.internal.Closer

Ex.
https://github.com/google/guava/blob/master/guava/src/com/google/common/io/Closer.java
https://github.com/google/guava/blob/master/guava-tests/test/com/google/common/io/CloserTest.java